pkg/generate/app/cmd/newapp.go

```diff
@@ -476,8 +476,63 @@ func (c *AppConfig) RunBuilds(out io.Writer) (*AppResult, error) {
 		return nil, ErrNoInputs
 	}
 
-	return c.run(out, app.Acceptors{app.NewAcceptBuildConfigs(c.typer), app.NewAcceptUnique(c.typer), app.AcceptNew},
+	bcAcceptor := app.NewAcceptBuildConfigs(c.typer)
+	result, err := c.run(out, app.Acceptors{bcAcceptor, app.NewAcceptUnique(c.typer), app.AcceptNew},
 		components, repositories, environment, parameters)
+	if err != nil {
+		return nil, err
+	}
+	return filterImageStreams(result), nil
+
+}
+
+func filterImageStreams(result *AppResult) *AppResult {
+	// 1st pass to get images from all BuildConfigs
+	imageStreams := map[string]bool{}
+	for _, item := range result.List.Items {
+		if bc, ok := item.(*buildapi.BuildConfig); ok {
+			to := bc.Parameters.Output.To
+			if to != nil && to.Kind == "ImageStreamTag" {
+				imageStreams[makeImageStreamKey(to)] = true
+			}
+			switch bc.Parameters.Strategy.Type {
+			case buildapi.DockerBuildStrategyType:
+				from := bc.Parameters.Strategy.DockerStrategy.From
+				if from != nil && from.Kind == "ImageStreamTag" {
+					imageStreams[makeImageStreamKey(from)] = true
+				}
+			case buildapi.SourceBuildStrategyType:
+				from := bc.Parameters.Strategy.SourceStrategy.From
+				if from.Kind == "ImageStreamTag" {
+					imageStreams[makeImageStreamKey(from)] = true
+				}
+			case buildapi.CustomBuildStrategyType:
+				from := bc.Parameters.Strategy.CustomStrategy.From
+				if from != nil && from.Kind == "ImageStreamTag" {
+					imageStreams[makeImageStreamKey(from)] = true
+				}
+			}
+		}
+	}
+	items := []runtime.Object{}
+	// 2nd pass to remove ImageStreams not used by BuildConfigs
+	for _, item := range result.List.Items {
+		if is, ok := item.(*imageapi.ImageStream); ok {
+			if _, ok := imageStreams[fmt.Sprintf("%s/%s", is.Namespace, is.Name)]; ok {
+				items = append(items, is)
+			}
+		} else {
+			items = append(items, item)
+		}
+	}
+	result.List.Items = items
+	return result
+}
+
+func makeImageStreamKey(ref *kapi.ObjectReference) string {
```

Comment:
We have imageapi.SplitImageStreamTag and JoinImageStreamTag for specifically this reason.  We have NamespacedName as well, use those three.